src/prpy/planning/cbirrt.py

```diff
@@ -265,8 +266,25 @@ def Plan(self, robot, smoothingitrs=None, timelimit=None, allowlimadj=0,
         # Bypass the context manager since CBiRRT does its own baking in C++.
         collision_checker = self.robot_checker_factory(robot)
         options = collision_checker.collision_options
-        with CollisionOptionsStateSaver(env.GetCollisionChecker(), options):
-            response = self.problem.SendCommand(args_str, True)
+        
+        if tsr_chains is not None:
+            mimicbodies = [env.GetKinBody(chain.mimicbodyname) 
+                           for chain in tsr_chains if chain.mimicbodyname is not 'NULL']
+            mimicbody_savers = [
+                mimicbody.CreateKinBodyStateSaver(KinBody.SaveParameters.LinkTransformation)
+                for mimicbody in mimicbodies]
+        else:
+            mimicbody_savers = []
+
+        upper, lower = robot.GetDOFLimits()
+
+        with CollisionOptionsStateSaver(env.GetCollisionChecker(), options), \
+            robot.CreateRobotStateSaver(Robot.SaveParameters.ActiveDOF | 
+                                        Robot.SaveParameters.LinkTransformation), \
+            contextlib.nested(*mimicbody_savers):
+            response = problem.SendCommand(args_str, True)
+
+        robot.SetDOFLimits(upper, lower)
```

Comment:
Put this in a `finally` block so it will run even if something raises an exception.